src/FINDOLOGIC/Export/Data/Price.php

```diff
@@ -6,8 +6,10 @@
 
 class Price extends UsergroupAwareSimpleValue
 {
+    const PRICE = true;
+
     public function __construct()
     {
-        parent::__construct('prices', 'price');
+        parent::__construct('prices', 'price', self::PRICE);
```

Comment:
As an alternative to passing around a constant, which could grow into a more awkward solution once more different types of data need validation, I propose a more OOP approach:

1.  Create a method `UsergroupAwareSimpleValue::validate($value)` that throws an exception in case of a bad value.
    * The method should be called in `UsergroupAwareSimpleValue::setValue()`.
    * The default implementation should just be empty, as it accepts any string.
    * Maybe consider moving the empty check into this method instead of leaving the default implementation empty.
2.  `Price` overrides the method and checks for the numeric nature of the value, in addition to calling the parent.
3.  Because there are more types of data requiring numeric values, perhaps we should also implement this for `SalesFrequency`, `Sort` and `Bonus`.
4.  If more than one thing gets the same extra validation, it's probably time to introduce `UsergroupAwareNumericValue`, that sits between `UsergroupAwareSimpleValue` and, for example `Price`. This removes duplicate code and potential for duplicate bugs.